conda_smithy/configure_feedstock.py

```diff
@@ -153,11 +189,15 @@ def merge_list_of_dicts(list_of_dicts):
     return squished_dict
 
 
-def argsort(seq):
+def argsort(
+    seq: List[tuple]
+) -> List[int]:
     return sorted(range(len(seq)), key=seq.__getitem__)
 
 
-def sort_config(config, zip_key_groups):
+def sort_config(
+    config: OrderedDict, zip_key_groups: List[Union[List[str], Any]]
```

Comment:
Is the type `Union[T, Any]` equivalent to `Any`? If not, what makes it different and how are you allowed to use the value that might be a `T` or anything else?

Reply:
adding the discussion from slack here

> https://neighbourhoodie.slack.com/archives/C069J3L39HN/p1718277724884569?thread_ts=1717776546.052819&cid=C069J3L39HN
if x: Union[A, B, C...], then for the call x.foo(), all types A, B, C etc need to have a foo() method
so that means Union[T, Any] and Any are not the same thing; it means a value could be anything but you need to wrap it in isinstance(val, T) before using it
and then you get the benefits of type-checking T, whereas if you'd just used Any you'd get no type checks at all